downsampler_plugin/algorithms/swinging_door_test.go

```
@@ -0,0 +1,594 @@
+					times = append(times, t0.Add(time.Duration(i)*100*time.Millisecond))
+				}
+
+				emitted := 0
+				for i, ts := range times {
+					val := 5.0 + 0.05*float64(i) // small changes, always below threshold
+					points, err := algo.Ingest(val, ts)
+					Expect(err).NotTo(HaveOccurred())
+					if i == 0 {
+						Expect(points).To(HaveLen(1), "First point should be emitted")
+						emitted++
+					} else if i%1 == 0 && i > 0 {
```

golangci-lint (1.64.8)

SA4028: x % 1 is always zero

(staticcheck)